deepmd/dpmodel/utils/serialization.py

```
@@ -170,3 +176,202 @@ def convert_numpy_ndarray(x: Any) -> Any:
+        return cls("ListNode", children, data, variables)
+
+    def __str__(self) -> str:
+        elbow = "└──"
+        pipe = "│  "
+        tee = "├──"
+        blank = "   "
+        linebreak = "\n"
+        buff = []
+        buff.append(f"{self.name} (size={format_big_number(self.size)})")
+        children_buff = []
+        for ii, (kk, vv) in enumerate(self.children.items()):
```

Ruff (0.14.13)

Loop control variable `ii` not used within loop body

Rename unused `ii` to `_ii`

(B007)